migrations/2025-05-06-145536_local_image_person/up.sql

```diff
@@ -0,0 +1,31 @@
+-- Since local thumbnails could be generated from posts of external users,
+-- use the person_id instead of local_user_id for the LocalImage table.
+--
+-- Also connect the thumbnail to a post id.
+--
+-- See https://github.com/LemmyNet/lemmy/issues/5564
+ALTER TABLE local_image
+    ADD COLUMN person_id int NOT NULL DEFAULT 0 REFERENCES person (id) ON UPDATE CASCADE ON DELETE CASCADE,
+    ADD COLUMN thumbnail_and_post_id int REFERENCES post (id) ON UPDATE CASCADE ON DELETE CASCADE;
```

Comment:
The name is a bit strange, why not simply post_id?

Reply:
I thought about that, but since the api will return this field as part of `local_image`, I think its better to make it very clear that this column means its both a thumbnail and this is the post id. 

I've also already added a comment about it too.

Reply:
what about `thumbnail_for_post_id`?

Reply:
@Nutomic lmk if you still want this done, otherwise it can be merged.

Reply:
Yes its better to rename, `thumbnail_for_post_id` sounds good.